src/runtime/coro.go

```diff
@@ -0,0 +1,31 @@
+package runtime
+
+// A naive implementation of coroutines that supports
+// package iter.
+
+type coro struct {
+	f  func(*coro)
+	ch chan struct{}
+}
+
+//go:linkname newcoro
+
+func newcoro(f func(*coro)) *coro {
```

Comment:
It doesn't look like this should work?
The syntax is `//go:linkname <functionname> <linkname>` and should be immediately before the function.

Reply:
Yeah, I just copied those lines from Big Go's coro.go. I'll leave them.